### PR TITLE
EventGrid rename RedisImportRDB to RedisImportRdb

### PR DESCRIPTION
**Pull request by YijunXieMS**

### CODE REVIEW

File: sdk/eventgrid/azure-messaging-eventgrid/src/main/java/com/azure/messaging/eventgrid/EventGridEvent.java [hunk 1]
```diff
@@ -269,4 +284,36 @@ public String getDataVersion() {
...
+            } catch (IOException e) {
+                throw LOGGER.logExceptionAsError(new UncheckedIOException(e));
+            }
+        }
+
+        @Override
+        public Mono<Void> serializeAsync(OutputStream stream, Object value) {
+            return Mono.fromRunnable(() -> serialize(stream, value));
+        }
+    }
```
Comment (srnagar) [inline sdk/eventgrid/azure-messaging-eventgrid/src/main/java/com/azure/messaging/eventgrid/EventGridEvent.java:317]: This is also in azure-core. For the next release, we should avoid this duplication.

File: sdk/eventgrid/azure-messaging-eventgrid/swagger/README.md [hunk 2]
```diff
@@ -68,6 +68,14 @@ directive:
     - rename-model:
         from: ResourceWriteSuccessData
         to: ResourceWriteSuccessEventData
+    - rename-model:
+        from: RedisImportRDBCompletedEventData
+        to: RedisImportRdbCompletedEventData
+    - rename-model:
+        from: RedisExportRdbCompletedEventData
+        to: RedisExportRDBCompletedEventData
```
Comment (srnagar) [inline sdk/eventgrid/azure-messaging-eventgrid/swagger/README.md:76]: Looks like these are inverted. `from` should have `*RDB*` and `to` should have `*Rdb*`.

**Review by srnagar — APPROVED** [review]

File: sdk/eventgrid/azure-messaging-eventgrid/src/main/java/com/azure/messaging/eventgrid/EventGridEvent.java (hunk 1 above)
Comment (YijunXieMS) [reply sdk/eventgrid/azure-messaging-eventgrid/src/main/java/com/azure/messaging/eventgrid/EventGridEvent.java:317]: Yes, I'll create a follow-up issue to have a default JsonSerializer in `azure-core`.

File: sdk/eventgrid/azure-messaging-eventgrid/swagger/README.md (hunk 2 above)
Comment (YijunXieMS) [reply sdk/eventgrid/azure-messaging-eventgrid/swagger/README.md]: reversed.

File: sdk/eventgrid/azure-messaging-eventgrid/README.md [hunk 3]
```diff
@@ -165,6 +165,11 @@ String sasToken = EventGridPublisherClient
     .generateSas("<your event grid endpoint>", new AzureKeyCredential("<key for the endpoint>"), expiration);
 ```
 
+### Use `BinaryData`
+`com.azure.core.util.BinaryData.fromObject(Object object)` and `toObject()` need a default Json serializer if you don't provide one.
```
Comment (srnagar) [inline sdk/eventgrid/azure-messaging-eventgrid/README.md:169]:
```suggestion
BinaryData supports serializing and deserializing objects through `com.azure.core.util.BinaryData.fromObject(Object object)` and `toObject()` methods. These methods need a default Json serializer in the classpath. Please include [azure-core-serializer-json-jackson](https://github.com/Azure/azure-sdk-for-java/tree/master/sdk/core/azure-core-serializer-json-jackson)
in your project `pom.xml` so `BinaryData` has a default json serializer.

You can also provide your own implementation of `ObjectSerializer` and use `fromObject(Object object, ObjectSerializer customSerializer)` or `toObject(Class<T> clazz, ObjectSerializer serializer)`

**Review by srnagar — APPROVED** [review]

**Pull request merged**